### PR TITLE
grpc server [4]: replace TCP with gRPC.

### PR DESCRIPTION
**Pull request by disksing**
The plan:
1. [x] update kvproto.
2. [x] remove kv/coprocessor from TCP server.
3. [x] impl  grpc server.
4. [x] impl raft grpc client.
5. [x] impl grpc snapshot.
6. [x] replace TCP server with grpc servers.
7. [ ] add tests.

### CODE REVIEW

File: tests/raftstore/server.rs [hunk 1]
```diff
@@ -251,14 +239,15 @@ impl Simulator for ServerCluster {
     }
 
     fn send_raft_msg(&self, raft_msg: raft_serverpb::RaftMessage) -> Result<()> {
+        println!("send raft msg type: {:?}",
```
Comment (siddontang) [inline tests/raftstore/server.rs]: remove it

File: src/server/metrics.rs [hunk 2]
```diff
@@ -41,12 +41,6 @@ lazy_static! {
             &["type"]
         ).unwrap();
 
-    pub static ref CONNECTION_GAUGE: Gauge =
```
Comment (siddontang) [inline src/server/metrics.rs]: so we can't know the connection count now?

Comment (disksing) [inline src/server/metrics.rs]: Yes, maybe we can patch grpc-rust to support it.

Comment (siddontang) [inline src/server/metrics.rs]: @overvenus maybe we can add it later.

**Pull request closed without merging**